### PR TITLE
Design game API schema

### LINKED ISSUE #54: Design game API request/response structs
Twirp support in C# isn't good. gRPC does not work on M1 Macs. So we're just going to define JSON manually...

This issue will definitely need group consensus before merging anything.

Includes:
- verification requests/responses steps
- starting/stopping experiments or rounds depending...
- recording "frames" / player data

### PR DESCRIPTION
**Pull request by danielholmes839**
closes #54

### CODE REVIEW

**Comment by danielholmes839** [comment]
Review https://github.com/seg491X-team36/virtual-spatial-navigation/pull/62/files#diff-9fda009ae19cccf75ef8003ccc5a3b56cf33a984652bf30488ee91b2ec5789d4R9-R99

**Review by FarinaSalman — APPROVED** [review]

File: docs/experiment-state-diagram.drawio [hunk 1]
```diff
@@ -0,0 +1 @@
...Rk0ztXaDWBG2qwexag1lvM2r2ZfNA+wHPBi0HPMfQK8GxSUJJe8DAA56tLlKvkvZMLmy74z09Iw+EUpYBjJ9lUCeqkoxSa8sX2Mlzliu0pdX2A6cO4865VPb/gG4ENIYcAZ1LCqg9mxZj2mE27Sd8kHM87sBZ26L9NUv+i6vCWRKpnL4Zn0idE1r6b7V1fV6BsXbi+jw3Xwuk6VMwAXUFOVze3J/flblCKt9PftdRg16K5oRyhlhQGCLWIXcc4KfLisXCA9DD1GAZZ8j4yr0e1SN4UXw1Hsm0ncftZo/g8yeFGso1cQ5xRqO48i2MQxsUrSEoRp05bkR0dySMA5R2O5E0zGca5NBtouOCYT2OcVcvv4QLxluQ5bTOtRy7IOu4bfsn5EWefP6sNabgWCfyODtV01wq/73KeR15n9REkzEZdF7H0WWzmBEwqPNQ+OLWqnlf12kHW29erbNjCXdIKCqwUxdxnys8FpCsylT3hJZ7ewfJQwKVsE59E7zkaUwGHJbAYLmN40DdnS/sxU0fd6gCJ+e+A9UtKGZe8sFqz0glX3oz5AikSRrXHnnsAuo+4PYwvhHQ5M+jlSCOBprGT/yOuZ/4Vpf/SPslHY1LOOzqH1cdnnz2kPr+etW00J49f+eKHJATg5oNmYOuAXLVdeTquq1PJEXJjnVbmdDDT4XoXQb2jxX/kPBtM6IrpEziEX5LIyKcfH0Yx2LHytrPNq74IpRAa0i5i4N5yQW/xXt28Cgucr3BDBdPFIfiYw4B3CLKyyY+WQX8yowXFbqQ74ZZw+A7FxUh5oIBzE75EVrzmmhORPxiwEW2xnCBCxWW3gqKG6y5LKZkxp6yFQ8QyVPxsHLGJn9BD6ZQtIAIbW/EulkXJHIfZG1iB3OfkKRsBpynKxiIB4bkiYtJueEnqxhxC5KGNQZdDSgNdjpfpWVVGfbmqFrhXkJTYl50NAfBQ9muQ4FK3hFcvkpEEaEIFEFvO16T6ittau1irBOTSCC0xsLu+MYriCOpygfOFTBXhqFakVlZsqKZOZBVmr3xKtR8/MSYics1nYael+PAjQPnW0FxpaVVT+z/rEpNDVIU72rCwbHg9S8FlScopppknKP7RI4lfzW4s5Fgoo7h0t5fRmeIRbremxmt5eSOpXGFtVPKL3C02Gn17f1siq/6Dwb2w/8=</diagram></mxfile>
```
Comment (FarinaSalman) [inline docs/experiment-state-diagram.drawio:1]: do we need this file?

Comment (danielholmes839) [reply docs/experiment-state-diagram.drawio:1]: should I add it to google drive instead?

Comment (FarinaSalman) [reply docs/experiment-state-diagram.drawio:1]: Yeah, I think the .PNG should stay but the .drawio file should be moved to google drive

**Pull request merged**